connect/runtime/src/test/java/org/apache/kafka/connect/runtime/TransformationStageTest.java

```diff
@@ -38,33 +39,36 @@ public class TransformationStageTest {
     private final SourceRecord transformed = new SourceRecord(singletonMap("transformed", 2), null, null, null, null);
 
     @Test
-    public void apply() {
+    public void apply() throws Exception {
         applyAndAssert(true, false, transformed);
         applyAndAssert(true, true, initial);
         applyAndAssert(false, false, initial);
         applyAndAssert(false, true, transformed);
     }
 
-    private void applyAndAssert(boolean predicateResult, boolean negate,
-                                SourceRecord expectedResult) {
-
-        @SuppressWarnings("unchecked")
+    @SuppressWarnings("unchecked")
+    private void applyAndAssert(boolean predicateResult, boolean negate, SourceRecord expectedResult) throws Exception {
+        Plugin<Predicate<SourceRecord>> predicatePlugin = mock(Plugin.class);
         Predicate<SourceRecord> predicate = mock(Predicate.class);
         when(predicate.test(any())).thenReturn(predicateResult);
-        @SuppressWarnings("unchecked")
+        when(predicatePlugin.get()).thenReturn(predicate);
+        Plugin<Transformation<SourceRecord>> transformationPlugin = mock(Plugin.class);
         Transformation<SourceRecord> transformation = mock(Transformation.class);
+        if ((predicateResult && !negate) || (!predicateResult && negate)) {
+            when(transformationPlugin.get()).thenReturn(transformation);
+        }
         if (expectedResult == transformed) {
             when(transformation.apply(any())).thenReturn(transformed);
         }
```

Comment:
nit: Use the same condition as for `apply`, because they both happen together.
```suggestion
        if (expectedResult == transformed) {
            when(transformationPlugin.get()).thenReturn(transformation);
            when(transformation.apply(any())).thenReturn(transformed);
        }
```